src/internal/transactionenv/env.go

```diff
@@ -397,10 +397,12 @@ func (env *TransactionEnv) WithWriteContext(ctx context.Context, cb func(*Transa
 			Client:        pachClient,
 			ClientContext: pachClient.Ctx(),
 			Stm:           stm,
-			pfsPropagater: env.pfsServer.NewPropagater(stm),
 			txnEnv:        env,
 		}
-		txnCtx.commitFinisher = env.pfsServer.NewPipelineFinisher(txnCtx)
+		if env.pfsServer != nil {
```

Comment:
The auth service creates it's own transaction contexts to run `InTransaction` methods, so we need to support a transaction context with no PFS server.